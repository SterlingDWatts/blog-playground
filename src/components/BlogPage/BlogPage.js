import React, { Component } from "react";
import BlogBody from "./BlogBody/BlogBody";
import BlogNotFound from "./BlogNotFound/BlogNotFound";
import BlogListItem from "./BlogListItem/BlogListItem";
import BlogContext from "../BlogContext";
import "./BlogPage.css";

class BlogPage extends Component {
  static contextType = BlogContext;

  render() {
    const blogs = this.context.blogs;
    const blog = blogs.find(b => b.id === this.props.match.params.blogID);
    const blogBody = blog ? <BlogBody blog={blog} /> : <BlogNotFound />;
    const blogList =
      blogs.length > 1
        ? blogs
            .filter(b => b.id !== this.props.match.params.blogID)
            .slice(0, 2)
            .map(b => <BlogListItem blog={b} key={b.id} />)
        : " ";
    return (
      <div className="BlogPage">
        {blogBody}
        <ul>{blogList}</ul>
      </div>
    );
  }
}

export default BlogPage;
