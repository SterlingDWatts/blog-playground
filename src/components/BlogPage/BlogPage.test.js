import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import BlogPage from "./BlogPage";

describe(`BlogPage Component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <BlogPage />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <BlogPage />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
