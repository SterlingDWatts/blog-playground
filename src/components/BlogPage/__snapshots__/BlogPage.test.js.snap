// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage Component renders the UI as expected 1`] = `
<div
  className="BlogPage"
>
  <nav
    className="NavBar__row NavBar__top NavBar__space-between NavBar__elpheba NavBar"
  >
    <div
      className="NavBar__container"
    >
      <button
        className="NavBarMarque"
        type="button"
      >
        <a
          className="elpheba"
          href="/blog"
          onClick={[Function]}
        >
          Sterling | Blog
        </a>
      </button>
      <button
        className="NavBarCollapse__hamburger_button elpheba"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars fa-w-14 "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <div
        className="NavBarCollapse__big"
      >
        <nav
          className="NavBarNavigation__s NavBarNavigation"
        >
          <ul
            className="NavBarNavigation__list"
          >
            <li>
              <div
                className="NavBarNavigation__item"
              >
                <a
                  className="NavBarNavItem elpheba"
                  href="/#about-me"
                  onClick={[Function]}
                >
                  About Me
                </a>
              </div>
            </li>
            <li>
              <div
                className="NavBarNavigation__item"
              >
                <a
                  className="NavBarNavItem elpheba"
                  href="/#projects"
                  onClick={[Function]}
                >
                  Projects
                </a>
              </div>
            </li>
            <li>
              <div
                className="NavBarNavigation__item"
              >
                <a
                  className="NavBarNavItem elpheba"
                  href="/#connect"
                  onClick={[Function]}
                >
                  Connect
                </a>
              </div>
            </li>
            <li>
              <div
                className="NavBarNavigation__item"
              >
                <a
                  className="NavBarNavItem elpheba"
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </div>
            </li>
          </ul>
        </nav>
      </div>
       
    </div>
  </nav>
  <main
    className="BlogNotFound"
  >
    <header>
      <h2>
        Blog Not Found
      </h2>
    </header>
  </main>
  <ul>
     
  </ul>
</div>
`;
